### ISSUE TITLE
Build issues related to upcoming @prenda/spark-icons

### ISSUE THREAD

**Issue by WilliamKelley** [prenda-school/prenda-spark#137]
#136 discussed one build issue for solely `@prenda/spark-icons`, but I was able to resolve that.

Some more have popped up:
- [x] Chromatic deployments fail ([runner output](https://github.com/prenda-school/prenda-spark/runs/3032957513?check_suite_focus=true)). 
  - ~The only info I can see as to why it fails is the bundle size being too large. The ideal scenario is that it would treeshake whatever it doesn't need, but even though `spark-icons/package.json` marks `"sideEffects": false`, that's not happening. Maybe because it's not built, and if we mapped imports from `'@prenda/spark-icons'` to point to the `dist/spark-icons/src`, it would tree-shake. Not sure.~
  - The above was not the cause for the error -- the use of `@ts-expect-error` was
  - An alternative is copying all spark icons we use into an internal folder of spark / the story folder.
  - *SOLUTION: The issue has been resolved in #134. The alternative above is still an appealing use-case, I'll move to another issue.*
- [x] Building spark fails
  - output
```bash
> npx nx run spark:build

> nx run spark:build 
Bundling spark...
Error during bundle: Error: /home/william/projects/spark/libs/spark/src/Select.ts(1,29): semantic error TS2307: Cannot find module '@prenda/spark-icons/ChevronDown' or its corresponding type declarations.
Bundle failed: spark
Error during bundle: Error: /home/william/projects/spark/libs/spark/src/Select.ts(1,29): semantic error TS2307: Cannot find module '@prenda/spark-icons/ChevronDown' or its corresponding type declarations.
Bundle failed: spark

———————————————————————————————————————————————

>  NX   ERROR  Running target "spark:build" failed

  Failed tasks:
  
  - spark:build

> npx nx run spark:build

> nx run spark:build 
Bundling spark...
Error during bundle: Error: /home/william/projects/spark/libs/spark/src/Select.ts(1,29): semantic error TS2307: Cannot find module '@prenda/spark-icons/ChevronDown' or its corresponding type declarations.
Bundle failed: spark
Error during bundle: Error: /home/william/projects/spark/libs/spark/src/Select.ts(1,29): semantic error TS2307: Cannot find module '@prenda/spark-icons/ChevronDown' or its corresponding type declarations.
Bundle failed: spark

———————————————————————————————————————————————

>  NX   ERROR  Running target "spark:build" failed

  Failed tasks:
  
  - spark:build
```
  - Similar error to what i was getting in #136 after a bit, but I don't know how to resolve this. I tried moving the aliasing in `libs/spark/.storybook/webpack.config.js` up to `.storybook/webpack.config.js` but that didn't work.
  - *SOLUTION: setting paths patterns in tsconfig and webpack to allow for correct solution of deep imports.*

**Comment by WilliamKelley** [comment]
We can solve the second issue about building Spark by making the import a named import
```tsx
import { ChevronDown as ChevronDownIcon } from '@prenda/spark';
```
And letting `@prenda/spark-icons` be a peer-dependency...

**Comment by WilliamKelley** [comment]
Solved both in #134, closing
